src/struct.h

```diff
@@ -56,4 +63,26 @@ struct Tuple
 };
 
 std::ostream &operator<<(std::ostream &os, const TupleFields &t);
+
+class StructManager
+{
+public:
+  using StructMap = std::map<std::string, std::shared_ptr<Struct>>;
+
+  void Add(const std::string &name, size_t size);
+  void AddField(const std::string &struct_name,
+                const std::string &field_name,
+                const SizedType &type,
+                ssize_t offset,
+                bool is_bitfield,
+                const Bitfield &bitfield,
+                bool is_data_loc);
```

Comment:
Wouldn't it make more sense for `AddField` be a method on `Struct` and not `StructManager`?

Reply:
Yeah, that makes more sense, fixed.